src/main/java/org/folio/circulation/domain/ConfigurationRepository.java

```diff
@@ -25,17 +33,37 @@ public ConfigurationRepository(Clients clients) {
       .thenApply(result -> result.map(relatedRecords::withTimeZone));
   }
 
+  public CompletableFuture<Result<Integer>> lookupSchedulerNoticesProcessingLimit() {
+    Result<CqlQuery> cqlQueryResult = defineCqlQueryResult("NOTIFICATION_SCHEDULER", "noticesLimit");
+    return lookupConfigurations(cqlQueryResult, applySearchSchedulerNoticesLimit());
+  }
+
   private CompletableFuture<Result<DateTimeZone>> findTimeZoneConfiguration() {
-    final ConfigurationService configurationService = new ConfigurationService();
+    Result<CqlQuery> cqlQueryResult = defineCqlQueryResult("ORG", "localeSettings");
+    return lookupConfigurations(cqlQueryResult, applySearchDateTimeZone());
+  }
+
+  private <T> CompletableFuture<Result<T>> lookupConfigurations(Result<CqlQuery> cqlQueryResult,
+                                                                Function<MultipleRecords<Configuration>, T> searchStrategy) {
 
-    final Result<CqlQuery> moduleQuery = exactMatch("module", "ORG");
-    final Result<CqlQuery> configNameQuery = exactMatch("configName", "localeSettings");
+    return cqlQueryResult
+      .after(query -> configurationClient.getMany(query, DEFAULT_PAGE_LIMIT))
+      .thenApply(result -> result.next(response -> from(response, Configuration::new, CONFIGS_KEY)))
+      .thenApply(result -> result.map(searchStrategy));
+  }
+
+  private Result<CqlQuery> defineCqlQueryResult(String moduleName, String configName) {
```

Comment:
What is this a CQL result for?

If it is for filtering by module and config names, maybe that could be more explicit in the name
```suggestion
  private Result<CqlQuery> defineModuleNameAndConfigNameFilter(String moduleName, String configName) {
```

Reply:
Yes, you're right, the method should display the correct name. Fix